modules/appeals_api/spec/workers/appeal_received_job_spec.rb

```diff
@@ -0,0 +1,219 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+# rubocop:disable Metrics/ModuleLength
+module AppealsApi
+  RSpec.describe AppealReceivedJob, type: :job do
+    describe 'higher_level_review' do
+      it 'errors if the keys needed are missing' do
+        opts = {
+          'receipt_event' => 'hlr_received'
+        }
+        expect(Rails.logger).to receive(:error).with 'AppealReceived: Missing required keys'
+
+        described_class.new.perform(opts)
+      end
+
+      it 'logs error if email identifier cannot be used' do
+        client = instance_double(VaNotify::Service)
+        allow(VaNotify::Service).to receive(:new).and_return(client)
+        opts = {
+          'receipt_event' => 'hlr_received',
+          'email_identifier' => { 'id_value' => 'fake_email@email.com' }, # missing id_type
+          'first_name' => 'first name',
+          'date_submitted' => DateTime.new(2021, 11, 11, 1, 2, 3).iso8601,
+          'guid' => '1234556'
+        }
+
+        expect(Rails.logger).to receive(:error)
```

Comment:
Just to cover our bases, let's make sure that we do not talk to VANotify in these error checking specs (especially since we set up an unsued double in these tests already). Right now it's just checking that we've logged the error, not that it's actually broken out of the run as expected.